src/openrct2/world/tile_inspector.c

```diff
@@ -82,7 +93,12 @@ sint32 tile_inspector_insert_corrupt_at(sint32 x, sint32 y, sint16 elementIndex,
 		// this way it's placed under the selected element, even when there are multiple elements with the same base height
 		for (sint16 i = 0; i < elementIndex; i++)
 		{
-			map_swap_elements_at(x, y, i, i + 1);
+			if (!map_swap_elements_at(x, y, i, i + 1)) {
+				// don't return error here, we've already inserted an element
+				// and moved it as far as we could, the only sensible thing left
+				// to do is to invalidate the window.
+				break;
```

Comment:
Since `map_get_nth_element_at` uses `elementIndex` and has already been passed, this break will never be hit.